File: bip-beneficios-backend/src/main/java/com/bip/backend/model/BeneficioModel.java
package com.bip.backend.model;

import com.bip.ejb.Beneficio;
import java.math.BigDecimal;
import lombok.Value;

@Value
public class BeneficioModel {

    Long id;
    String nome;
    String descricao;
    BigDecimal valor;
    boolean ativo;
    Long version;

    public static BeneficioModel fromEntity(Beneficio beneficio) {
        return new BeneficioModel(
                beneficio.getId(),
                beneficio.getNome(),
                beneficio.getDescricao(),
                beneficio.getValor(),
                beneficio.isAtivo(),
                beneficio.getVersion()
        );
    }
}

File: bip-beneficios-frontend/src/app/modules/beneficios/models/beneficio.model.ts
export interface Beneficio {
  id: number;
  nome: string;
  descricao: string | null;
  valor: number;
  ativo: boolean;
  version: number | null;
}

export interface BeneficioPayload {
  nome: string;
  descricao: string | null;
  valor: number;
  ativo: boolean;
}

export interface TransferenciaPayload {
  origemId: number;
  destinoId: number;
  valor: number;
}

export interface TransferenciaResultado {
  origemId: number;
  destinoId: number;
  valorTransferido: number;
  saldoOrigem: number;
  saldoDestino: number;
}
